src/foraging_gui/TransferToNWB.py

```diff
@@ -38,8 +38,17 @@ def _get_field(obj, field_list, reject_list=[], index=None, default=np.nan):
         field_list = [field_list]
         
     for f in field_list:
-        if hasattr(obj, f):
-            value = getattr(obj, f)
+            tag=0
+            if type(obj) is type:
+                if hasattr(obj, f):
+                    value = getattr(obj, f)
+                    tag=1
+            elif type(obj) is dict:
+                if f in obj:
+                    value = obj[f]
+                    tag=1
```

Comment:
can you add doc string for this part?

Reply:
Updated

# the obj.Opto_dialog is a dictionary